lnrpc/invoicesrpc/invoices.proto

```diff
@@ -135,3 +141,11 @@ message SubscribeSingleInvoiceRequest {
     // Hash corresponding to the (hold) invoice to subscribe to.
     bytes r_hash = 2;
 }
+
+message LookupInvoiceMsg {
+    oneof invoice_ref {
+        bytes payment_hash = 1;
+        bytes payment_addr = 2;
+        bytes set_id = 3;
```

Comment:
`set_id` is chosen by the sender. What happens when two users choose the same set_id to pay to different amp invoices?

Reply:
Right now we assume it's globally unique across _all_ invoices: https://github.com/lightningnetwork/lnd/blob/master/channeldb/invoices.go#L1906

It's a 32 byte value, so the odds of this happening are the same as two different services picking the same payment hash which would cause a payment to fail at the _sender_ once the second payment was attempted.

Reply:
I was merely wondering if it can be used to do something malicious. But looking at the code it seems that the second set with the same id will just be rejected.